static/app/views/organizationIntegrations/integrationListDirectory.tsx

```diff
@@ -224,8 +224,15 @@ export class IntegrationListDirectory extends AsyncComponent<
     return integrations?.find(i => i.provider.key === integration.key) ? 2 : 0;
   }
 
-  getPopularityWeight = (integration: AppOrProviderOrPlugin) =>
-    POPULARITY_WEIGHT[integration.slug] ?? 1;
+  getPopularityWeight = (integration: AppOrProviderOrPlugin) => {
+    for (const sentryapp of this.state.publishedApps || []) {
```

Comment:
I'm sure there is some very cool and fancy one line way to do this that I can't figure out

Reply:
👀 
```js
const fallback = POPULARITY_WEIGHT[integration.slug] ?? 1
return this.state.publishedApps.find(app => app === integration)?.popularity ?? fallback
```